### ISSUE TITLE
[Help] <chatglm2-6b loss 下降问题>

### ISSUE THREAD

**Issue by Huffman-cotdom** [THUDM/ChatGLM2-6B#249]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

训练数据量：210W
训练方式：deepspeed stag2（官方的DeepSpeed配置）
loss: 
  - epoch0
    - step40: 9.8429 -> step80: 9.6052 -> step910: 8.6776  -> step4180: 7.558
  - epoch1
    - step4280: 7.2281 -> step5710: 5.9984
想问一下大家在微调时loss下降趋势是怎么样的？我这边一开始loss下降有点慢，不清楚是数据的问题，还是代码有bug

### Steps To Reproduce

官方 chatglm-6b 运行脚本：
LR=1e-4
python3 -u -m torch.distributed.run \
    --nnodes 2 \
    --node_rank 0 \
    --nproc_per_node 8 \
    --master_addr ${METIS_WORKER_0_HOST} \
    --master_port ${MASTER_PORT} \
    main.py \
    --deepspeed deepspeed.json \
    --do_train \
    --train_file ${data_path}/train.json \
    --test_file ${data_path}/dev.json \
    --prompt_column content \
    --glm_version 2 \  # glm2 计算context_length 有问题修改了代码
    --response_column summary \
    --overwrite_cache \
    --model_name_or_path ${model_path} \
    --output_dir ${ckpt_path}/adgen-chatglm2-6b-ft-$LR \
    --overwrite_output_dir \
    --max_source_length 512 \
    --max_target_length 512 \
    --per_device_train_batch_size 8 \
    --per_device_eval_batch_size 1 \
    --gradient_accumulation_steps 4 \
    --predict_with_generate \
    --max_steps 20000 \
    --logging_steps 10 \
    --save_steps 500 \
    --learning_rate $LR \
    --fp16 \

### Environment

```markdown
- OS: centos
- Python: 3.9
- Transformers: 4.28.1
- PyTorch: 2.10
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) : True
```
